controllers/awsmachine_controller.go

```diff
@@ -186,6 +214,20 @@ func (r *AWSMachineReconciler) SetupWithManager(mgr ctrl.Manager, options contro
 			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.AWSClusterToAWSMachines)},
 		).
 		WithEventFilter(pausedPredicates(r.Log)).
+		WithEventFilter(
+			predicate.Funcs{
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					if e.ObjectOld.GetObjectKind().GroupVersionKind().Kind != "AWSMachine" {
+						return true
+					}
+
+					oldMachine := e.ObjectOld.(*infrav1.AWSMachine)
+					newMachine := e.ObjectNew.(*infrav1.AWSMachine)
+
+					return !reflect.DeepEqual(oldMachine.Spec, newMachine.Spec)
```

Comment:
Wondering if this can be simplified/generalized by looking at `medata.ObservedGeneration`

Reply:
yep it can be. From what I see, CAPA doesn't have any of the types updated to include ObservedGeneration yet

Reply:
Unless we have a driving reason to add this predicate checking, I think we should likely avoid it for now. At least until we better understand potential implications of it.